### ISSUE TITLE
[BUG] AttributeError: 'NoneType' object has no attribute 'bands' in `DataCollection`

### ISSUE THREAD

**Issue by mlubej** [sentinel-hub/eo-learn#263]
**Describe the bug**

Bug discovered yesterday with @AleksMat, creating the ticket just to have it written down. 

In certain cases of running the `SentinelHubInputTask` you might get an error saying 

```
AttributeError: 'NoneType' object has no attribute 'bands'
```

Pointing to the `data_collection` being `None` in [this line](https://github.com/sentinel-hub/eo-learn/blob/09285d22fc6cf33af359fbaf85c5426e9a2d09f2/io/eolearn/io/processing_api.py#L256)

The solution is adding a missing `self`, like below:

```python
bands = self.data_collection.bands
```

**Comment by AleksMat** [comment]
We'll fix this in the next release. However, a simple way to solve this is to instead of `data_source` use `data_collection` parameter. Parameter `data_source` is deprecated anyways.

**Comment by AleksMat** [comment]
Fixed in `0.9.2`
